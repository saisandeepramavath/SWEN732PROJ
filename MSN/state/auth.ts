import {atom} from 'recoil'

import { FirebaseAuthTypes } from '@react-native-firebase/auth';

export const authState = atom<{
  isAuth: boolean;
  user: FirebaseAuthTypes.User | null;
}>({
  key: 'authState',
  default: {
    isAuth: false,
    user: null,
  },
});

export const userState = atom({
    key: 'userState',
    default: {
        name: '',
        email: '',
        photo: '',
        uid: '',
        groups: [],
    }
})

export const groupsState = atom({
    key: 'groupsState',
    default: [],
})
export const selectedGroupState = atom({
    key: 'selectedGroupState',
    default: null,
})
export const selectedGroupMembersState = atom({
    key: 'selectedGroupMembersState',
    default: [],
})
export const selectedGroupTransactionsState = atom({
    key: 'selectedGroupTransactionsState',
    default: [],
})