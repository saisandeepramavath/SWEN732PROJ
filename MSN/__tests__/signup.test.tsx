import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import Signup from '../app/signup';
import auth from '@react-native-firebase/auth';
import firestore from '@react-native-firebase/firestore';


describe('Signup', () => {
	it('should match snapshot', () => {
		const { toJSON } = render(<Signup />);
		expect(toJSON()).toMatchSnapshot();
	});

	it('should display error when passwords do not match', async () => {
		const { getByPlaceholderText, getByTestId, findByTestId } = render(<Signup />);
		fireEvent.changeText(getByPlaceholderText('Full Name'), 'John Doe');
		fireEvent.changeText(getByPlaceholderText('Email'), 'john@example.com');
		fireEvent.changeText(getByPlaceholderText('Phone Number'), '1234567890');
		fireEvent.changeText(getByPlaceholderText('Password'), 'password123');
		fireEvent.changeText(getByPlaceholderText('Confirm Password'), 'password456');
		fireEvent.press(getByTestId('signup-button'));

		// âœ… Now looking for testID instead of text
		expect(await findByTestId('error-message')).toHaveTextContent('Passwords do not match');
	});

  it('should call auth and firestore on successful signup', async () => {
    const { getByPlaceholderText, getByTestId } = render(<Signup />);
    
    fireEvent.changeText(getByPlaceholderText('Full Name'), 'John Doe');
    fireEvent.changeText(getByPlaceholderText('Email'), 'john@example.com');
    fireEvent.changeText(getByPlaceholderText('Phone Number'), '1234567890');
    fireEvent.changeText(getByPlaceholderText('Password'), 'password123');
    fireEvent.changeText(getByPlaceholderText('Confirm Password'), 'password123');
  
    // Ensure button press is triggered
    fireEvent.press(getByTestId('signup-button'));
    console.log('ðŸš€ Button Pressed in Test');
  
    // Wait for authentication to be called
    // await waitFor(() => {
    //   expect(auth().createUserWithEmailAndPassword).toHaveBeenCalledTimes(1);
    //   expect(auth().createUserWithEmailAndPassword).toHaveBeenCalledWith('john@example.com', 'password123');
    //   expect(firestore().collection).toHaveBeenCalledWith('users');
    //   // expect(firestore().collection('users').doc).toHaveBeenCalledWith(0);
    //   expect(firestore().collection('users').doc('123').set).toHaveBeenCalledWith({
    //     fullName: 'John Doe',
    //     email: 'john@example.com',
    //     phoneNumber: '1234567890',
    //   });
    // });
  });
  
  
});
