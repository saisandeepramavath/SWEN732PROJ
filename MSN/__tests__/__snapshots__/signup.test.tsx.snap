// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 32,
        "fontWeight": "bold",
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Signup
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Full Name"
    placeholderTextColor="#aaa"
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#ddd",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 50,
        "marginVertical": 10,
        "padding": 10,
      }
    }
    value=""
  />
  <TextInput
    autoCapitalize="none"
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="#aaa"
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#ddd",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 50,
        "marginVertical": 10,
        "padding": 10,
      }
    }
    value=""
  />
  <TextInput
    keyboardType="phone-pad"
    onChangeText={[Function]}
    placeholder="Phone Number"
    placeholderTextColor="#aaa"
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#ddd",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 50,
        "marginVertical": 10,
        "padding": 10,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    placeholderTextColor="#aaa"
    secureTextEntry={true}
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#ddd",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 50,
        "marginVertical": 10,
        "padding": 10,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Confirm Password"
    placeholderTextColor="#aaa"
    secureTextEntry={true}
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#ddd",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 50,
        "marginVertical": 10,
        "padding": 10,
      }
    }
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="signup-button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        disabled={false}
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Signup
      </Text>
    </View>
  </View>
</View>
`;
